service/src/main/java/org/cbioportal/service/util/MutationMapperUtils.java

```diff
@@ -31,7 +47,60 @@
+            structuralVariant.setEventInfo(fusion.getProteinChange());
+            if (fusion.getProteinChange() != null) {
+                String proteinChange = fusion.getProteinChange();
+                // only parse proteinChange when its not Fusion or SV
+                if (!(proteinChange.equalsIgnoreCase("Fusion") || proteinChange.equalsIgnoreCase("SV"))) {
+
+                    Matcher matcher = r.matcher(proteinChange);
+                    String site1GeneSymbol = null;
+                    String site2GeneSymbol = null;
+                    VariantType variantType = null;
+
+                    if (matcher.find()) {
+                        if (EnumUtils.isValidEnum(VariantType.class, matcher.group(1).toUpperCase())) {
+                            // if protein change contains only variant type. ex: INTRAGENIC
+                            variantType = EnumUtils.getEnum(VariantType.class, matcher.group(1).toUpperCase());
+
+                        } else if (EnumUtils.isValidEnum(VariantType.class, matcher.group(2).toUpperCase())) {
+                            // this is in format of <gene>-<variant-type>. ex: TUFT1-intragenic
+                            site1GeneSymbol = matcher.group(1);
+                            variantType = EnumUtils.getEnum(VariantType.class, matcher.group(2).toUpperCase());
+                        } else {
+                            // this is in format of <gene1>-<gene2>-<optional variant-type>. ex.
+                            // ZSWIM4-SLC1A6 or ZNF595-TERT fusion
+                            site1GeneSymbol = matcher.group(1);
+                            site2GeneSymbol = matcher.group(2);
+                            if (EnumUtils.isValidEnum(VariantType.class, matcher.group(3).toUpperCase())) {
+                                variantType = EnumUtils.getEnum(VariantType.class, matcher.group(3).toUpperCase());
+                            }
+                        }
+
+                        // only set site2Gene if its not null
+                        if (site2GeneSymbol != null) {
+                            if (site1GeneSymbol != null && site1GeneSymbol.equalsIgnoreCase(site2GeneSymbol)) {
+                                structuralVariant.setSite2EntrezGeneId(fusion.getEntrezGeneId());
+                                structuralVariant.setSite2HugoSymbol(fusion.getGene().getHugoGeneSymbol());
+                            } else {
+                                try {
+                                    Gene site2Gene = geneService.getGene(site2GeneSymbol);
+                                    if (site2Gene != null) {
+                                        structuralVariant.setSite2EntrezGeneId(site2Gene.getGeneticEntityId());
+                                        structuralVariant.setSite2HugoSymbol(site2Gene.getHugoGeneSymbol());
+                                    }
+                                } catch (Exception e) {
```

Comment:
Is this when a gene symbol is not found?

Reply:
Yes. I'll add a comment here

Reply:
This will go away once we move to a more structured SV format.